### PR TITLE
Add HTTPRoute stubs

### PR DESCRIPTION
**Pull request by andrewstucki**
### Changes proposed in this PR:

THIS IS A STACKED PR SINCE IT REQUIRESR https://github.com/hashicorp/consul-api-gateway/pull/338 TO BE MERGED FIRST.

This re-orgs a bit of the handler structure and adds some stub routes for HTTPRoute CRUD operations.

### Checklist:
- [x] Tests added
- [ ] CHANGELOG entry added 
  > Run `make changelog-entry` for guidance in authoring a changelog entry, and
  > commit the resulting file, which should have a name matching your PR number.
  > Entries should use imperative present tense (e.g. Add support for...)

### CODE REVIEW

**Review by jherschman — APPROVED** [review]
Looking goood 👍

File: internal/api/schemas/v1.yaml [hunk 1]
```diff
@@ -271,6 +439,178 @@ components:
...
+      properties:
+        name:
+          type: string
+        namespace:
+          type: string
+        weight:
+          type: number
+        filters:
+          "$ref": "#/components/schemas/HTTPFilters"
+    HTTPFilters:
```
Comment (nathancoleman) [inline internal/api/schemas/v1.yaml:517]: Worth going ahead and adding path prefix replace filter since we support it in K8s land now?

**Review by nathancoleman — APPROVED** [review]

File: internal/api/schemas/v1.yaml [hunk 2]
```diff
@@ -271,6 +439,178 @@ components:
...
+          minLength: 1
+        namespace:
+          type: string
+          minLength: 1
+    HTTPRoute:
+      type: object
+      required:
+      - name
+      - namespace
+      - gateways
```
Comment (nathancoleman) [inline internal/api/schemas/v1.yaml:470]: Analogous to `ParentRefs`, K8s doesn't require these be set. Thoughts?

https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1alpha2.CommonRouteSpec

File: internal/api/schemas/v1.yaml (hunk 1 above)
Comment (andrewstucki) [reply internal/api/schemas/v1.yaml:517]: Good call, I'll do that in a follow-up.

File: internal/api/schemas/v1.yaml (hunk 2 above)
Comment (andrewstucki) [reply internal/api/schemas/v1.yaml:470]: I'd prefer to keep this as required, at least initially. I've never quite understood why a route doesn't need a gateway in the K8s API -- think maybe due to the initial thoughts around things like route delegation?

Either way, it'd be a breaking change to all of a sudden disallow empty gateways, but it's a lot easier to make the validation _less_ strict.

Comment (mikemorris) [reply internal/api/schemas/v1.yaml:470]: Yea, spec flexibility will be useful here in the future for route delegation, GAMMA mesh binding, etc - requiring binding to a gateway for now makes sense to me.

**Pull request merged**